File: C++/uasqueue.cpp
#include <iostream>
#include <string>
#define MAX_Q 10
using namespace std;

struct Queue {
   string data[MAX_Q];
   int head;
   int tail;
};
Queue antrian;

void create() {
    antrian.head= antrian.tail = -1;
}

int isEmpty() {
    if (antrian.tail == -1){
        return 1;
    } else {
        return 0;
    }
}

int isFull() {
    if (antrian.tail == MAX_Q-1){
        return 1;
    } else {
        return 0;
    }
}

void Enqueue(string data) {
    if(isEmpty()==1){
        antrian.head=antrian.tail=0;
        antrian.data[antrian.tail]=data;
        cout << "Antrian Masuk" << antrian.data[antrian.tail];
    } else if(isFull()==0){
            antrian.tail++;
            antrian.data[antrian.tail]=data;
            cout << "Antrian Masuk" << antrian.data[antrian.tail];
    }
}

string Dequeue() {
    int i;
    string e = antrian.data[antrian.head];
    for(i=antrian.head; i<=antrian.tail-1; i++){
        antrian.data[i] = antrian.data[i+1];
    }
    antrian.tail--;
    return e;
}

void Clear(){
    antrian.head=antrian.tail=0;
    cout << "Antrian Sudah Dihapus Semua" << endl;
}
void inQueue(){
    if (!isEmpty()) {
        cout << "Dalam Antrian : " << endl;
        for (int i = 0; i < antrian.tail; i++) {
            cout << i+1 << " " << antrian.data[i+1] << endl;
        }
    }else {
        cout << "Antrian Kosong" << endl;
    }

    if(isFull()) {
        cout << "Antrian Sedang Penuh" << endl;
    }
}

void tmbQueue(){
    string antri;
    cout << "Nama Pesanan : ";
    cin.ignore();
    getline(cin, antri);
    Enqueue(antri);
    cout<<"Antrian ditambah"<< endl;
}

int main(){ 
    int hall=0;
    do {
        system("cls");
        cout << "Dapur Masak Pesanan Makan" << endl;
        inQueue();
        cout << endl;
        cout << "Menu: 1.Tambah Antrian, 2.Antrian Sudah Selesai, 3.Hapus Semua Antrian 4. Mengakhiri Program : ";
        cin >> hall;   
        switch (hall)  {
            case 1: tmbQueue(); break;
            case 2: Dequeue();  break;
            case 3: Clear(); break;
            case 4: cout << "Terima Kasih!" << endl; return 0;
            default: cout << "null" << endl; break;
        }
    }
    while (true);
}

File: java/uasqueue.java
package uas.java;

// iki dicopy ngisor e pakage
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

/**
 * 1. hapus kabeh seng public class seng pakage karo autor ditokne
 * 
 */

 // iki kabeh dicopy
 // uasqueue iki dihanti karo jeneng file mu contone BinaryTree kemau gede cilik e podo
public class uasqueue {
    static Queue<String> antrian = new LinkedList<>();

    public static void main(String[] args) {
        int hall = 0;
        do {
            System.out.println("Dapur Masak Pesanan Makan");
            inQueue();
            System.out.println();
            System.out.print("Menu: 1.Tambah Antrian, 2.Antrian Sudah Selesai, 3.Hapus Semua Antrian, 4. Mengakhiri Program: ");
            Scanner scanner = new Scanner(System.in);
            hall = scanner.nextInt();

            switch (hall) {
                case 1:
                    tmbQueue();
                    break;
                case 2:
                    Dequeue();
                    break;
                case 3:
                    Clear();
                    break;
                case 4:
                    System.out.println("Terima Kasih!");
                    return;
                default:
                    System.out.println("Pilihan tidak valid!");
                    break;
            }
        } while (true);        
    }

    public static void inQueue() {
        if (!antrian.isEmpty()) {
            System.out.println("Dalam antrian : ");
            int i = 1;
            for (String pesanan : antrian) {
                System.out.println(i + " " + pesanan);
                i++;
            }
        } else {
            System.out.println("Antrian Kosong");
        }

        if (antrian.size() >= 10) {
            System.out.println("Antrian Sedang Penuh");
        }
    }

    public static void tmbQueue() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Nama Pesanan : ");
        String antri = scanner.nextLine();
        if (antrian.size() < 10) {
            antrian.offer(antri);
            System.out.println("Antrian ditambah");
        } else {
            System.out.println("Antrian sudah penuh! Tidak dapat menambahkan pesanan.");
        }
    }

    public static void Dequeue() {
        if (!antrian.isEmpty()) {
            String pesanan = antrian.poll();
            System.out.println("Pesanan " + pesanan + " selesai");
        } else {
            System.out.println("Antrian kosong! Tidak ada pesanan yang selesai.");
        }
    }

    public static void Clear() {
        if (!antrian.isEmpty()) {
            antrian.clear();
            System.out.println("Semua antrian telah dihapus");
        } else {
            System.out.println("Antrian sudah kosong");
        }
    }
    
}
